src/libopensc/card-epass2003.c

```diff
@@ -323,8 +324,10 @@ openssl_dec(const EVP_CIPHER * cipher, const unsigned char *key, const unsigned
 	ctx = EVP_CIPHER_CTX_new();
 	if (ctx == NULL)
 		goto out;
-	EVP_DecryptInit_ex(ctx, cipher, NULL, key, iv_tmp);
-	EVP_CIPHER_CTX_set_padding(ctx, 0);
+
+	if (!EVP_DecryptInit_ex(ctx, cipher, NULL, key, iv_tmp)
+			|| EVP_CIPHER_CTX_set_padding(ctx, 0))
```

Comment:
this should be negated. If you put the || on the first line, clang should not complain about it:
```suggestion
	if (!EVP_DecryptInit_ex(ctx, cipher, NULL, key, iv_tmp) ||
			EVP_CIPHER_CTX_set_padding(ctx, 0) != 1)
```

Reply:
Yes, you're just too quick reviewing...